<!DOCTYPE html>
<html>
<body>
	{% extends "layout.html" %}
	{% block content %}
	<div id="column_l" style="width: 700px">
		<h1>Temporal Ordering of Sampled Data Readings for Biological Data</h1>
		<p>This is a web-based application that makes available different techniques to order data samples temporally.
		Any of these methods can be used for data that has been sampled at various points in time, where the exact time point of sampling in unknown. 
		this can be used in the following way:
		</p>
		<ul>
			<li>Upload file containing data samples, without sampling times, in a valid format</li>
			<li>Select the temporal ordering approach to be used on the input data</li>
			<li>Obtain an ordering of the data samples. The ordering is given in the form of numbers for the data samples that is based on their order in the 
			input file. The results are displayed graphically as well</li>
		</ul>
		
		<p>These methods can be used for biological data such as gene expressions, mRNA profiling experiments,
		cell lineage tracing. Although the algorithms implemented here were primarily constructed for biological data, it can find applications in 
		any kind data that follows a chronological order. For instance, archeological samples.
		This is because, the main principle behind all these methods, is to find the distance or level of dissimilarity between data points, and use that 
		to predict the closeness of the sampled data points with respect to time. This is assuming the data is for a process that progresses or changes as
		a function of time.
		</p>
		
		<form method=post enctype=multipart/form-data target="_blank">
      		<p><input type=file name=file multiple><br>
      		Select file type uploaded:<br>
      		<tr>
      			<td><input type="radio" name="options" id="option1" value="option1"> Tab Separated </input></td>
      			<td><input type="radio" name="options" id="option2" value="option2"> Comma Separated </input></td>
      			<td><input type="radio" name="options" id="option3" value="option3"> Fasta file (Aligned Sequences) </input></td>
      			<td><input type="radio" name="options" id="option4" value="option4"> Multiple Host files (Quasi species sequences) </input></td>
      		</tr><br>
      		
      		Choose a label option:<br>
      		<tr>
      			<td><input type="radio" name="labeloptions" id="option3" value="nolabel"> Without Labels </input></td>
      			<td><input type="radio" name="labeloptions" id="option4" value="label"> With Labels </input></td>
      		</tr><br>
			

      		<tr>
      			<td><input type="radio" name="methodoptions" id="option8" value="all" checked="checked" style="display:none"/></td>
      		</tr><br>
			
			
        	<input type=submit value="Upload and compare methods"></p>
    	</form>
		
	</div>
	<div id="column_r" >
		<h1>Sample File Downloads</h1>
		<p>The input file provided can be a tab separated file, comma separated file, or a sequence fasta file. The accepted file extensions are
		.txt, .csv, .fas, .fa, .fasta. If the input file is a genetic sequence file, it should have a file extension of .fas, .fa, or .fasta. 
		Each new line of the file indicates the biological variable recorded for all instances of sampling. 
		For numerical tab separated or comma separated data sets, each tab or comma separated value in every line will 
		denote an observation/sampling done in an unknown time point.</p>
		<ul> 
			<li><a href="{{ url_for('download_file', filename='jellyroll.txt') }}">Artificial Jellyroll Dataset</a></li>
			<p>Artificial two dimensional dataset consisting 150 points. Magwene <i>et al.</i>,(2003)</p>
			<li><a href="{{ url_for('download_file', filename='data.csv') }}">Cell cycle data</a></li>
			<p>Gene expression data matrix of 3196 high variance genes across 17 unordered samples from only one cell cycle. Dataset was  
			obtained from the supplemental text for Qiu <i>et al</i>,(2011)</p>
		</ul>
		<p><a href="{{ url_for('visualize') }}" target="_blank">Visualize results for sample dataset - Jellyroll</a></p>
		
	</div>
	</div>   
	{% endblock %}
</body>
</html>